File: Medium/215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> pQ = new PriorityQueue<>();
        for(int i:nums){
            pQ.add(i);
            if(pQ.size()>k) pQ.remove();
        }
        return pQ.remove();
    }
}

File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        int n = nums.size();

        if (k <= n / 2) {
            priority_queue<int, vector<int>, greater<int>> minHeap;
            for (int num : nums) {
                minHeap.push(num);
                if (minHeap.size() > k) {
                    minHeap.pop();
                }
            }
            return minHeap.top();
        } else {
            // Use max heap of size (n - k + 1) to find (n-k+1)th smallest
            // element
            int newK = n - k + 1;
            priority_queue<int> maxHeap;
            for (int num : nums) {
                maxHeap.push(num);
                if (maxHeap.size() > newK) {
                    maxHeap.pop();
                }
            }
            return maxHeap.top();
        }
    }
};